File: src/main/java/gg/mineral/practice/managers/TournamentManager.java
package gg.mineral.practice.managers;

import gg.mineral.practice.tournaments.Tournament;
import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;

import java.util.Map;

public class TournamentManager {
    private final static Map<String, Tournament> tournaments = new Object2ObjectOpenHashMap<>();

    public static void registerTournament(Tournament tournament) {
        tournaments.put(tournament.getHost(), tournament);
    }

    public static void remove(Tournament tournament) {
        tournaments.remove(tournament.getHost());
    }

    public static Tournament getTournament(String name) {
        return tournaments.get(name);
    }
}

File: src/main/kotlin/gg/mineral/practice/managers/TournamentManager.kt
package gg.mineral.practice.managers

import gg.mineral.practice.tournaments.Tournament
import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap

object TournamentManager {
    private val tournaments: MutableMap<String, Tournament> = Object2ObjectOpenHashMap()

    fun registerTournament(tournament: Tournament) {
        tournaments[tournament.host] = tournament
    }

    fun remove(tournament: Tournament) = tournaments.remove(tournament.host)

    fun getTournament(name: String) = tournaments[name]
}
